src/components/Interactive.js

```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash/fp'
+import { forwardRef, useState } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import * as Utils from 'src/libs/utils'
+
+
+const allowedHoverVariables = ['backgroundColor', 'color', 'boxShadow', 'opacity', 'textDecoration']
+const pointerTags = ['button', 'area', 'a', 'select']
+const pointerTypes = ['radio', 'checkbox', 'submit', 'button']
+
+const Interactive = forwardRef(({
+  className = '', as, type, role, onClick, onKeyDown, onMouseDown, onBlur, disabled, children, tabIndex, hover = {}, style = {}, ...props
+}, ref) => {
+  const [outline, setOutline] = useState(undefined)
+  const { cursor } = style
+
+  const computedCursor = Utils.cond(
+    [cursor, cursor],
+    [disabled, undefined],
+    [onClick || pointerTags.includes(as) || pointerTypes.includes(type), 'pointer']
+  )
+
+  const computedTabIndex = Utils.cond(
+    [_.isNumber(tabIndex), tabIndex],
+    [disabled, undefined],
+    [onClick, 0],
+    undefined)
+
+  const computedRole = Utils.cond(
+    [role, role],
+    [onClick && !['input', ...pointerTags].includes(as), 'button'],
+    undefined)
+
+  const cssVariables = _.flow(
+    _.toPairs,
+    _.flatMap(([key, value]) => {
+      console.assert(allowedHoverVariables.includes(key),
+        `${key} needs to be added to the hover-style in style.css for the style to be applied`)
+      return [[`--app-hover-${key}`, value], [key, `var(--hover-${key}, ${style[key]})`]]
+    }),
+    _.fromPairs
+  )(hover)
+
+  return h(as, {
+    ref,
+    className: `hover-style ${className}`,
+    style: { ...style, ...cssVariables, cursor: computedCursor, outline },
+    onKeyDown: onKeyDown || (e => e.key === 'Enter' && onClick && onClick(e)),
```

Comment:
One more issue I just found: This will call the `onClick` handler on this component, but will not bubble up to higher up components the way an actual click would. Some components (popup menus) are relying on that behavior. We might need to fire an actual browser event here.